packages/kbn-es/src/cluster.js

```diff
@@ -558,4 +569,113 @@ exports.Cluster = class Cluster {
     }
     return esJavaOpts.trim();
   }
+
+  /**
+   * Common initial setup for running Docker and Serverless
+   */
+  async setupDocker() {
+    if (this._process || this._outcome) {
+      throw new Error('ES has already been started');
+    }
+
+    await verifyDockerInstalled(this._log);
+    await maybeCreateDockerNetwork(this._log);
+  }
+
+  /**
+   * Run an Elasticsearch Docker container
+   *
+   * @param {DockerOptions} options
+   */
+  async runDocker(options = {}) {
+    await this.setupDocker();
+
+    const dockerCmd = resolveDockerCmd(options);
+
+    this._log.info(chalk.dim(`docker ${dockerCmd.join(' ')}`));
+    this._process = await execa('docker', dockerCmd);
+    // TODO: logging
+    // TODO: not detached?
+  }
+
+  /**
+   * Run an Elasticsearch Serverless Docker cluster
+   *
+   * @param {ServerlessOptions} options
+   */
+  async runServerless(options = {}) {
+    await this.setupDocker();
+
+    const volumeParentPath = await setupServerlessVolumes(this._log, options);
+    const volumeCmd = ['--volume', `${volumeParentPath}:/objectstore:z`];
+    const image = resolveDockerImage({
+      ...options,
+      repo: SERVERLESS_REPO,
+      defaultImg: SERVERLESS_IMG,
+    });
+
+    await runServerlessEsNode(this._log, {
+      params: [
+        '-p',
+        '9200:9200',
```

Comment:
Thoughts on prefixing these ports?
```suggestion
        '127.0.0.1:9200:9200',
```
Docker's will publish these routes, including rewriting firewall rules by default otherwise.  
https://docs.docker.com/network/#published-ports